src/lib.rs

```diff
@@ -735,6 +740,47 @@ impl Device {
 
         None
     }
+
+    #[cfg(feature = "log")]
+    /// Prints the PCI configuration space in a structured format similar to `lspci -x`.
+    ///
+    /// # Example Output
+    /// ```
+    /// 00:19.00 Ethernet Vendor:10EC Device:8168 (rev 10)
+    /// 00: 1a bc 3d e4 ...
+    /// 10: 4f 5a 1b 3c ...
+    /// ```
+    pub fn print_pci_config_space(&self) {
+        let mut buf = ArrayString::<300>::new();
+
+        write!(
+            buf,
+            "\n{:02X}:{:02X}.{:02X} {:?} Vendor:{:02X} Device:{:02X} (rev {})",
+            self.bdf.bus.val(),
+            self.bdf.dev.val(),
+            self.bdf.func.val(),
+            self.device_info.desc,
+            self.vendor_id,
+            self.device_id,
+            self.revision,
+        )
+        .unwrap();
+
+        for i in 0..64 {
+            let value = if cfg!(test) {
+                0x0 + i % 0xff
+            } else {
+                self.cfg_read16(i as u16)
+            };
+
+            if i % 16 == 0 {
+                write!(buf, "\n{:02x}: ", i).unwrap();
+            }
+            write!(buf, "{:02x} ", value).unwrap();
+        }
+        write!(buf, "\n").unwrap();
```

Comment:
Since this code is in a library, I would rather not have it unwrap if it could fail. Since these are just log statements anyway, could we explicitly ignore the result of each write!() call? Like
```
let _ = write!("{}", foo);
```